healthx/handler_test.go

```diff
@@ -36,6 +36,15 @@ import (
 	"github.com/ory/herodot"
 )
 
+const mockHeaderKey = "middleware-header"
+const mockHeaderValue = "test-header-value"
+
+func WithMiddleware(h func(http.Handler) http.Handler) func(*Options) {
+	return func(o *Options) {
+		o.Middleware = h
+	}
+}
```

Comment:
why in the `_test` file instead of inside the `handler`?

Reply:
Only needed it in the test
might act as an example for anyone who would want to implement something similar 
also, so that I wouldn't have to combine it all in one function for the test
 at least, that was my chain of thought